### DISCUSSION TITLE
Problems with the world browser

### DISCUSSION THREAD

**Discussion by JackTheFoxOtter** [Yellow-Dog-Man/Resonite-Issues#1399] [General]
Currently, I see a number of problems with the world browser that result in user confusion and frustration. I want to collect and discuss common usability pain-points here, before creating individual issues about them.

**Comment by JackTheFoxOtter** [comment]
### Opened Sessions
It is not intuitive to understand which sessions are currently opened. They are sorted to the top and have a red "X" in the corner, however they are also still affected by the current filter, which can make them disappear from the grid alltogether. I would prefer the active sessions to be displayed separately from the browser / search grid itself, and clearly labeled as "Opened Sessions" or "Worlds you're in", with a descriptive "Leave" or "Close" button, instead of just an "X" in the corner.

**Reply by JackTheFoxOtter** [reply]
For a bit more context, I've met a LOT of new users that didn't understand you can be in multiple sessions at the same time, had the 4 different sessions they've been to prior still open in the background, and were complaining about performance issues (which went away for the most part once I explainted to them how they can close open sessions). This is something that's not intuitive to understand, as it is not a feature people are familiar with from other platforms, so improving this is what I believe would help the most.

**Comment by JackTheFoxOtter** [comment]
### World / Session Info Panel
The two-level menu structure for clicking on a world / session from the browser grid, over opening the small preview, to then expanding to the full panel is confusing. I've met many new users that didn't even realize that there is a second layer to the menu, because the arrow in the corner is rather non-descriptive and confusing. Replacing the icon there would help, but I think it might make more sense to replace this menu with a single level.

**Comment by JackTheFoxOtter** [comment]
### Search Improvements
The search currently doesn't tell you about the search syntax. Clicking on the search bar could expand it to show a panel briefly describing how to use the search. I'd also like search parameters like tags to be visually highlighted in the search prompt. Being able to enter more advanced search commands (similar to GitHub's search) would be nice. If the possible search commands are displayed below the search bar, clicking them should automatically add them to the current search prompt (to reduce the amount of typing required for VR users).

**Comment by JackTheFoxOtter** [comment]
### Search Filters
The default filters are useful, and the custom filter is nice, even if using it feels a little confusing. It would be nice if users were able to name and save custom filters (so you can click a saved one without opening the customization menu every time), or even edit the default ones (although if the latter is possible it should probably also be possible to easily reset the filters to default). If the search syntax is expanded, the custom filter configuration panel might not even be necessary anymore. A button near the search bar for "Save current search" could replace it, with the fitler just auto-filling the current search prompt when you click on it.

**Reply by Lexevolution** [reply]
Another pain point for me with the search filters is that Local worlds (worlds that are saved to your local home) [don't appear in the "My Worlds" filter](https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/911).

![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/31176843/b4f76987-ed3f-4335-a7c5-8410bb67b324)

![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/31176843/3fb12fec-77d5-4b90-b531-708e2802da9a)

I believe they should since... they are your own world. Even though it's technically made by "a different user", My Worlds should include local machine worlds.
